### ISSUE TITLE
finetune coco caption got loss nan

### ISSUE THREAD

**Issue by maoxiaofei99** [salesforce/LAVIS#111]
hi, blip2 is very creative and impressive.
While i try to run official code with command "torchrun --nproc_per_node=8  train.py --cfg-path lavis/projects/blip2/train/caption_coco_ft.yaml",  i always get loss nan after one iteration with detail logs as below :
[INFO] Start training epoch 0, 4427 iters per inner epoch.
Train: data epoch: [0]  [   0/4427]  eta: 9:01:56  lr: 0.000000  loss: 1.9126  time: 7.3449  data: 0.0000  max mem: 26284
[INFO] Reducer buckets have been rebuilt in this iteration.
Train: data epoch: [0]  [  50/4427]  eta: 6:32:09  lr: 0.000001  loss: nan  time: 5.3553  data: 0.0000  max mem: 34957
Are there run-params in yaml file  that should be changed ?

**Comment by dxli94** [comment]
Please see https://github.com/salesforce/LAVIS/commit/0e9a3ecfa5dc668761ce6c0e3cf827578e8593c4 for the fix.

**Comment by dxli94** [comment]
Let us know how it goes. Feel welcome to reopen if issue persists.

**Comment by maoxiaofei99** [comment]
hi, thanks for the updating, seems training works well with the fixed code,  so far it has run hundreds of iterations. will keep observe 😊

**Comment by ECE-Engineer** [comment]
> hi, blip2 is very creative and impressive. While i try to run official code with command "torchrun --nproc_per_node=8 train.py --cfg-path lavis/projects/blip2/train/caption_coco_ft.yaml", i always get loss nan after one iteration with detail logs as below : [INFO] Start training epoch 0, 4427 iters per inner epoch. Train: data epoch: [0] [ 0/4427] eta: 9:01:56 lr: 0.000000 loss: 1.9126 time: 7.3449 data: 0.0000 max mem: 26284 [INFO] Reducer buckets have been rebuilt in this iteration. Train: data epoch: [0] [ 50/4427] eta: 6:32:09 lr: 0.000001 loss: nan time: 5.3553 data: 0.0000 max mem: 34957 Are there run-params in yaml file that should be changed ?

Can I ask what notebook and/or tutorial you used for finetuning BLIP2?

I'm trying to do it on a custom dataset, but am having trouble finding a fine tuning notebook or set of instructions to fine tune BLIP2.

**Comment by matthewdm0816** [comment]
Hello, in this commit, the training code chose not to optimize the ViT image encoder part for finetuning, which is not the same as the finetuning setting in the paper, and leads to suboptimal result compared with the performance reported in the paper after finetuning (I experimented on VQA2.0 finetuning and got 70 v. 82 in the paper), while unfreezing ViT leads to the nan loss problem (following layer-wise learning-rate decay and other setting in paper). Freezing ViT do solve the problem but it corrupts the performance. 

Could you provide the training setting used to finetune on VQA2.0 (e.g., the config file as for COCO Captions [)](https://github.com/salesforce/LAVIS/blob/aff3008b75b64f151d2a3918c0419fc195072008/lavis/projects/blip2/train/caption_coco_ft.yaml)

> Fixed in [0e9a3ec](https://github.com/salesforce/LAVIS/commit/0e9a3ecfa5dc668761ce6c0e3cf827578e8593c4).
